it/controllers/TaxCreditsChoiceControllerItSpec.scala

```diff
@@ -0,0 +1,262 @@
+          .willReturn(aResponse().withStatus(NOT_FOUND))
+      )
+
+      val request = FakeRequest(GET, url)
+
+      val result = route(app, request)
+
+      result.get.futureValue.header.status mustBe SEE_OTHER
+
+      result.get.futureValue.header.headers.get("Location") mustBe Some("/personal-account/your-address/residential/do-you-live-in-the-uk")
+    }
+
+    "return a SEE_OTHER and redirect to the beginning of the PTA address change journey if the user skipped to tax credits url" in {
+
+      server.stubFor(
+        get(urlEqualTo(citizenDetailsUrl))
+          .willReturn(ok(FileHelper.loadFile("./it/resources/citizen-details.json")))
+      )
+
+      server.stubFor(
+        get(urlEqualTo(personDetailsUrl))
+          .willReturn(ok(FileHelper.loadFile("./it/resources/person-details.json")))
+      )
+
+      server.stubFor(
+        get(urlPathMatching(s"$cacheMap/.*"))
+          .willReturn(aResponse().withStatus(OK).withBody("""
+                                                            |{
+                                                            |	"id": "session-id",
+                                                            |	"data": {
+                                                            |   "addressPageVisitedDto": {
+                                                            |     "hasVisitedPage": true
+                                                            |   }
+                                                            |	},
+                                                            |	"modifiedDetails": {
+                                                            |		"createdAt": {
+                                                            |			"$date": 1400258561678
+                                                            |		},
+                                                            |		"lastUpdated": {
+                                                            |			"$date": 1400258561675
+                                                            |		}
+                                                            |	}
+                                                            |}
+                                                            |""".stripMargin)
+          )
+      )
+
+      server.stubFor(
+        get(urlEqualTo(tcsBrokerUrl))
+          .willReturn(aResponse().withStatus(NOT_FOUND))
+      )
+
+      val request = FakeRequest(GET, url)
+
+      val result = route(app, request)
+
+      result.get.futureValue.header.status mustBe SEE_OTHER
+
+      result.get.futureValue.header.headers.get("Location") mustBe Some("/personal-account/your-address/residential/do-you-live-in-the-uk")
+    }
+
+    List(
+      BAD_REQUEST,
+      IM_A_TEAPOT,
+      INTERNAL_SERVER_ERROR,
+      SERVICE_UNAVAILABLE
+    ).foreach { response =>
+      s"return a SEE_OTHER and redirect to PTA Address Change if the call to TCS fails with a $response" in {
```

Comment:
SEE_OTHER should be INTERNAL_SERVER_ERROR
everything after redirect should be change to be show a technical difficulty page.

Reply:
Done